### PR TITLE
Feat/key

### PR DESCRIPTION
**Pull request by amars29**

### CODE REVIEW

File: src/plugins/RendererPlugin/components/Renderer.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const Renderer: React.SFC<{
   return (
     <>
       {components.map((Component: any) => (
```
Comment (Ajamuar) [inline src/plugins/RendererPlugin/components/Renderer.tsx]: **Change (Blocking)**: Do not use any. 
**Suggestion**: Use better variable name such as `componentsDataArray`.
**Suggestion** Take component inside the map in a variable and then use that variable

File: src/plugins/RendererPlugin/index.tsx [hunk 2]
```diff
@@ -2,10 +2,14 @@ import { IPlugin } from '../../interfaces/IPlugin';
...
 import { Renderer } from './components/Renderer';
 import ComponentUpdatedEvent from './events/ComponentUpdatedEvent';
+import uuid from './uuid';
 
 export class RendererPlugin implements IPlugin {
   public pluginStore: PluginStore = new PluginStore();
-  private componentMap = new Map<string, Array<React.Component>>();
+  private componentMap = new Map<
+    string,
+    Array<{ component: React.Component; key?: string }>
```
Comment (Ajamuar) [inline src/plugins/RendererPlugin/index.tsx]: **Suggestion (non-blocking)**: Let's declare the type in types folder of `RendererPlugin` and use it from there.

File: src/plugins/RendererPlugin/index.tsx [hunk 3]
```diff
@@ -18,23 +22,40 @@ export class RendererPlugin implements IPlugin {
...
     this.pluginStore.dispatchEvent(
       new ComponentUpdatedEvent('Renderer.componentUpdated', position)
     );
   }
 
-  removeFromComponentMap(position: string, component: React.Component) {
+  removeFromComponentMap(
+    position: string,
+    component: React.Component,
+    key?: String
```
Comment (Ajamuar) [inline src/plugins/RendererPlugin/index.tsx]: **Suggestion (non-blocking):** key not needed here

File: src/plugins/RendererPlugin/index.tsx [hunk 4]
```diff
@@ -18,23 +22,40 @@ export class RendererPlugin implements IPlugin {
...
+  ) {
     let array = this.componentMap.get(position);
     if (array) {
-      array.splice(array.indexOf(component), 1);
+      key
```
Comment (Ajamuar) [inline src/plugins/RendererPlugin/index.tsx]: **Suggestion (non-blocking)**: Search only using component

File: src/plugins/RendererPlugin/uuid.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+var uuid = require('react-uuid');
```
Comment (Ajamuar) [inline src/plugins/RendererPlugin/uuid.ts]: **Change:** Never use `var`

**Review by Ajamuar — CHANGES_REQUESTED** [review]

**Review by himanshu-satija — APPROVED** [review]

**Pull request merged**